bin/publish-release

```diff
@@ -49,32 +56,45 @@ const { getCurrentBranch, pullLatest, checkoutBranch } = require('../release-src
 
 const currentBranch = getCurrentBranch();
 
-checkoutBranch('master');
-pullLatest();
+async function publish() {
+    checkoutBranch('master');
+    pullLatest();
 
-// @NB: to ensure latest version we need to get package.json version from master
-const packageJson = require('../package.json');
+    // @NB: to ensure latest version we need to get package.json version from master
+    const packageJson = require('../package.json'); // eslint-disable-line global-require
 
-const latestVersion = `v${packageJson.version}`;
+    const latestVersion = `v${packageJson.version}`;
 
-userMessage(`Publishing UI React ${latestVersion} to NPM`, 'magenta', { iconRight: '🚀' });
+    userMessage(`Publishing UI React ${latestVersion} to NPM`, 'magenta', { iconRight: '🚀' });
 
-shell.exec('git fetch --all --tags', executeSilently);
+    shell.exec('git fetch --all --tags', executeSilently);
 
-userMessage(`Checking out tag ${latestVersion}...`, 'yellow');
-shell.exec(`git checkout tags/${latestVersion}`, executeSilently);
+    userMessage(`Checking out tag ${latestVersion}...`, 'yellow');
+    shell.exec(`git checkout tags/${latestVersion}`, executeSilently);
 
-userMessage('Publishing to NPM...', 'yellow');
+    userMessage('Run build script...', 'yellow');
 
-if (hasNpmRc) {
-    npm.publish();
-} else {
-    npm.setAuthToken()
-        .then(npm.publish)
-        .catch(error => console.error(error));
-}
+    shell.exec('yarn build', executeSilently);
+
+    userMessage('Publishing to NPM...', 'yellow');
+
+    await publishToNPM();
+
+    userMessage(`Checking out original branch (${currentBranch})...`, 'yellow');
+    checkoutBranch(currentBranch);
 
-userMessage(`Checking out original branch (${currentBranch})...`, 'yellow');
-checkoutBranch(currentBranch);
+    userMessage(
+        `Successfully released version ${latestVersion}`,
+        'green',
+        { iconRight: '🎉' },
+    );
+
+    return 0;
+}
 
-return 0;
+try {
+    return publish();
+} catch (error) {
+    console.error(error);
+    return 1;
+}
```

Comment:
Why are we not using [ink](http://npm.im/ink) for this?

Reply:
Because we didn't know Ink existed!!!  And ooooh useful!

This originally started as a bash script, I then converted it into a JS script – we didn't want to spend too much time on it

If we get the opportunity to come back and improve we'll definitely look at Ink.

Thanks for the tip!